src/diff/index.js

```diff
@@ -208,12 +208,6 @@ export function diff(
 			}
 
 			c._force = false;
-		} else if (
-			excessDomChildren == null &&
-			newVNode._original === oldVNode._original
```

Comment:
This condition should never occur for non-function children

Reply:
This might be useful for people who want to declare a chunk of static markup outside of render once (perhaps using that static vnode hoisting babel plugin?) and just re-use the vnodes, but perhaps this is better done by things like htm flags, etc. Either I don't have a strong preference atm.

Reply:
FWIW the JSX hoisting thing is super common, it's enabled by default in `@babel/preset-react`.